admin/spec/lib/validator/check-window-validator.spec.js

```diff
@@ -34,95 +38,257 @@ describe('Check window validator', function () {
+      expect(validationError.hasError()).toBe(true)
+      done()
+    })
 
-  it('should require a check start month', async function (done) {
-    req.body = getBody()
-    req.body.checkStartMonth = ''
-    let validationError = await checkWindowValidator.validate(req)
-    expect(validationError.hasError()).toBe(true)
-    done()
-  })
+    it('should require a check start day', async function (done) {
+      req.body.checkStartDay = ''
+      const validationError = await checkWindowValidator.validate(req)
+      expect(validationError.hasError()).toBe(true)
+      done()
+    })
 
-  it('should require a check start year', async function (done) {
-    req.body = getBody()
-    req.body.checkStartYear = ''
-    let validationError = await checkWindowValidator.validate(req)
-    expect(validationError.hasError()).toBe(true)
-    done()
-  })
+    it('should require a check start month', async function (done) {
+      req.body.checkStartMonth = ''
+      const validationError = await checkWindowValidator.validate(req)
+      expect(validationError.hasError()).toBe(true)
+      done()
+    })
+
+    it('should require a check start year', async function (done) {
+      req.body.checkStartYear = ''
+      const validationError = await checkWindowValidator.validate(req)
+      expect(validationError.hasError()).toBe(true)
+      done()
+    })
+
+    it('should require a check end day', async function (done) {
+      req.body.checkEndDay = ''
+      const validationError = await checkWindowValidator.validate(req)
+      expect(validationError.hasError()).toBe(true)
+      done()
+    })
+
+    it('should require a check end month', async function (done) {
+      req.body.checkEndMonth = ''
+      const validationError = await checkWindowValidator.validate(req)
+      expect(validationError.hasError()).toBe(true)
+      done()
+    })
 
-  it('should require a check end day', async function (done) {
-    req.body = getBody()
-    req.body.checkEndDay = ''
-    let validationError = await checkWindowValidator.validate(req)
-    expect(validationError.hasError()).toBe(true)
-    done()
+    it('should require a check end year', async function (done) {
+      req.body.checkEndYear = ''
+      const validationError = await checkWindowValidator.validate(req)
+      expect(validationError.hasError()).toBe(true)
+      done()
+    })
   })
 
-  it('should require a check end month', async function (done) {
-    req.body = getBody()
-    req.body.checkEndMonth = ''
-    let validationError = await checkWindowValidator.validate(req)
-    expect(validationError.hasError()).toBe(true)
-    done()
+  describe('When editing a check window', function () {
+    it('should require a check window name when editing', async function (done) {
+      req.body.checkWindowName = ''
+      req.body.checkWindowId = 'abc123'
+      const validationError = await checkWindowValidator.validate(req)
+      expect(validationError.hasError()).toBe(true)
```

Comment:
how can we be sure that it's this exact field that has the error?  i would usually go for an explicit assertion such as `expect(validationError.message).toBe('check name is required')`